src/components/CheckboxInput/CheckboxInput.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { v4 as uuid } from 'uuid';
+import './CheckboxInput.scss';
+
+const CheckboxInput = ({
+  id,
+  className,
+  selected,
+  onChange,
+  isDisabled,
+  children,
+}) => {
+  const [isSelected, setIsSelected] = useState(selected || false);
+  const inputId = id || uuid();
+
+  const handleChange = () => {
+    setIsSelected(!isSelected);
+    onChange(!isSelected);
+  };
+
+  return (
+    <div className={cx('checkboxInput', { isDisabled })}>
+      <input
+        id={inputId}
+        checked={isSelected}
+        disabled={isDisabled}
+        onChange={handleChange}
+        type="checkbox"
+        className={className}
+        value={isSelected}
+        style={{ width: '24px', height: '24px' }}
+      />
+      <span className="checkboxInputInstructions">{children}</span>
```

Comment:
should be a <label> right?

Reply:
It should be a label, yes. But in the signup use case, it contains a link to view the terms and conditions. And, we don't want it to toggle the checkbox when the user does so. 

Not sure what to do about this one.

Reply:
modern browsers handle this correctly now (they don't toggle the checkbox when a nested link is clicked). 

here's a demo: https://codesandbox.io/s/angry-haslett-m6dl9?file=/index.html